processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -55,6 +61,18 @@ public static void verifyAggregations(
             "Missing fields [%s] for postAggregator [%s]", missing, postAgg.getName()
         );
         Preconditions.checkArgument(combinedAggNames.add(postAgg.getName()), "[%s] already defined", postAgg.getName());
+
+        if (postAgg instanceof HasDependentAggFactories) {
```

Comment:
I believe you are referring to testComputedInArithmeticPostAggregator. 
there its only testing computation instead of dependency injection. 
can you add/modify the test to call Queries.prepareAggregations() for arithmetic aggregator and verify that all dependencies are set properly ?